### PR TITLE
Fix for incompatible css property types

### PR DESCRIPTION
**Pull request by tutok**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
fix: https://github.com/deity-io/falcon/issues/545

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**

**Other information:**

### CODE REVIEW

File: packages/falcon-ui/src/components/Icon.tsx [hunk 1]
```diff
@@ -18,9 +19,11 @@ export const IconRenderer = themed({
...
 });
 
-type IconProps = { src: string; fallback?: any } & ThemedComponentProps & BaseProps<'svg'>;
-
-const IconComponent: React.SFC<IconProps & PropsWithTheme> = props => {
+export type IconProps = IconRendererProps & {
+  src: string;
+  fallback?: any;
+};
+const IconImpl: React.SFC<IconProps & PropsWithTheme> = props => {
```
Comment (Borales) [inline packages/falcon-ui/src/components/Icon.tsx]: let's not use `Impl` abbreviation in class names, please. I thought we discussed this part as well 😛 (after having `CheckoutImpl`). I thought `IconComponent` was ok for this.

Comment (tutok) [reply packages/falcon-ui/src/components/Icon.tsx]: hmm, we have already couple of `Impl` components

https://github.com/deity-io/falcon/blob/233c55d74f40387ad5362d994abecfeaddff92b4/packages/falcon-ui-kit/src/Responsive/Responsive.tsx#L32

https://github.com/deity-io/falcon/blob/6aa8c877ffc1d2001332ece8f866ff2f02fd8f1c/packages/falcon-front-kit/src/ScrollToTop/ScrollToTop.tsx#L20

that is why I also rename `IconComponent` into `IconImpl`

Comment (tutok) [reply packages/falcon-ui/src/components/Icon.tsx]: Ok, let's change all of them into `*Inner`

Comment (tutok) [reply packages/falcon-ui/src/components/Icon.tsx]: done

**Pull request merged**